test/metabase/test/data/datasets.clj

```
@@ -43,6 +43,11 @@
                     #{:h2})]
     (when config/is-test?
       (log/info (color/cyan "Running QP tests against these engines: " engines)))
+
+    (when-not (every? all-valid-engines engines)
```

This will fail fast if you specify an engine and that driver is not available.